File: design-patterns/go/adapter/transport/bike.go
// bike.go

package main

type Bike struct {
}

func NewBike() (bike *Bike) {
	bike = &Bike{}
	return
}

func (bike *Bike) GetNumberOfWheels() int {
	return 2
}

func (bike *Bike) GetWeight() float64 {
	return 700
}

func (bike *Bike) GetDistanceTravelled() float64 {
	return 80
}

func (bike *Bike) GetTravelCostPerMile() float64 {
	return 4
}

File: design-patterns/ts/adapter/transport/bike.ts
// bike.ts

import Transport from "./transport";

class Bike implements Transport {
    getNumberOfWheels(): number {
        return 2;
    }

    getWeight(): number {
        return 700;
    }

    getDistanceTravelled(): number {
        return 80;
    }

    getTravelCostPerMile(): number {
        return 4;
    }
}

export default Bike;